txnkv/transaction/2pc.go

```diff
@@ -1019,6 +1019,13 @@ func (c *twoPhaseCommitter) checkAsyncCommit() bool {
 		return false
 	}
 
+	// Don't use async commit when writing on cached table.
+	// The actual restriction is (cachedTableWriteLease < commitTS), so using async commit is possible.
+	// But that's unnecessary, because cached table is not designed for write performance.
+	if c.txn.cachedTableWriteLease != nil {
```

Comment:
Is it possible to make it more abstract and avoid using the term `table`? It does not have a clear meaning at kv layer.

Reply:
`writeLease` seems not a good idea.
Any suggestion about the naming?

Reply:
How about expose a method `SetCommitTSValidFunction` instead of `SetCachedTableWriteLease` to custom validation

Reply:
Now it's changed to `SetCommitTSUpperBoundCheck` which is much more general...